xbmc/games/controllers/guicontrols/GUIGameControllerList.cpp

```diff
@@ -193,3 +187,34 @@ void CGUIGameControllerList::UpdatePeripheral(const PERIPHERALS::PeripheralPtr&
 {
   m_peripheralLocation = agentPeripheral->Location();
 }
+
+std::string CGUIGameControllerList::GetPortAddress(
+    const PERIPHERALS::PeripheralPtr& agentPeripheral)
+{
+  CAgentInput& agentInput = CServiceBroker::GetGameServices().AgentInput();
+
+  // Upcast peripheral to input providers
+  KEYBOARD::IKeyboardInputProvider* const keyboardInputProvider =
+      static_cast<KEYBOARD::IKeyboardInputProvider*>(agentPeripheral.get());
+  MOUSE::IMouseInputProvider* const mouseInputProvider =
+      static_cast<MOUSE::IMouseInputProvider*>(agentPeripheral.get());
+  JOYSTICK::IInputProvider* const joystickInputProvider =
+      static_cast<JOYSTICK::IInputProvider*>(agentPeripheral.get());
+
+  // See if the keyboard input provider has a port address
+  const std::string keyboardAddress = agentInput.GetKeyboardAddress(keyboardInputProvider);
```

Comment:
Please remove the `const` here and for the following strings because it prevents a move. Clang warning: `[clang-tidy] (performance-no-automatic-move) Constness of 'keyboardAddress' prevents automatic move`

Reply:
thanks, updated